homeassistant/components/telegram_bot/__init__.py

```diff
@@ -857,131 +858,102 @@ def leave_chat(self, chat_id=None):
 class BaseTelegramBotEntity:
     """The base class for the telegram bot."""
 
-    def __init__(self, hass, allowed_chat_ids):
+    def __init__(self, hass, config):
         """Initialize the bot base class."""
-        self.allowed_chat_ids = allowed_chat_ids
+        self.allowed_chat_ids = config[CONF_ALLOWED_CHAT_IDS]
         self.hass = hass
-
-    def _get_message_data(self, msg_data):
-        """Return boolean msg_data_is_ok and dict msg_data."""
-        if not msg_data:
-            return False, None
-        bad_fields = (
-            "text" not in msg_data and "data" not in msg_data and "chat" not in msg_data
+        # Dispatcher is set up in platform subclass
+        self.dispatcher.add_handler(  # pylint: disable=no-member
```

Comment:
The dispatch attribute should be made an `abstractmethod` property that needs to be implemented by subclasses.

Reply:
I felt just moving that line to the respective subclass yielded more readable code.